### PR TITLE
Upgraded to go 1.17

### LINKED ISSUE #10: Support for darwin-arm64 in further releases
It would be nice to have support for darwin-amd64 in forthcoming releases - so the provider can run on newer macOS versions on Apple M1 silicon.

Please kindly consider adding support for that OS/architecture combination.

### PR DESCRIPTION
**Pull request by sergei-ivanov**
Upgraded go dependencies
Fixes #10

Signed-off-by: Sergei Ivanov <sergei_ivanov@mail.ru>

### CODE REVIEW

**Comment by sergei-ivanov** [comment]
Notes:
1. I haven't updated the changelog / release notes, because I am not sure how they are maintained in this project.
2. The latest version of `golangci-lint` throws a few new errors when run via Makefile, but they seem to be unimportant, and I did not fix them. Mostly because I have a 99% ready branch with a major refactoring for #8 and I'll probably address some of them there.
3. One outstanding linting issue concerns deprecation of `golang.org/x/crypto`. We may need to look for a supported alternative (there are forks by ProtonMail and Keybase). Probably should be addressed as part of #9

File: README.md
Comment (invidian) [inline README.md]: This is enforced by `go.mod` file change.

File: .tool-versions
Comment (invidian) [inline .tool-versions]: Hmm, I'm not fan of optional tooling configuration inside the repository, as they are not as commonly used (lack of standard). This is why there is a `Dockerfile` and CI, which enforces the commonly agreed versions where builds etc must pass.

 Perhaps we could add a CI job instead which use specific versions of tooling then. I can take care of this. But CI should also be moved to GH actions actually... :smile:

**Review by invidian — CHANGES_REQUESTED** [review]
Just some nits, overall LGTM. Thanks for contribution!

File: go.mod
Comment (invidian) [inline go.mod:3]: For darwin-arm64 support, we will be fine bumping it to 1.16, no need to 1.17. 1.16 is still supported, so it should be fine to keep it.

Comment (sergei-ivanov) [reply go.mod:3]: Any reason why not use the current version of go? Just curious, because there are no upstream or downstream dependencies that might require backward compatibility.

File: .tool-versions
Comment (sergei-ivanov) [reply .tool-versions]: I suspected that it might not be accepted :( It just made my life a bit easier,
a) because it has all tool versions in one place, instead of them being scattered around config files, and 
b) because we use asdf+direnv that allows us to switch between projects easily.
So whenever I change into this project's folder, I have the right version of the tools in $PATH.

Sigh, I'll remove it.

File: README.md
Comment (sergei-ivanov) [reply README.md]: I thought we needed some additional clarity about that, but will remove.

File: .tool-versions
Comment (invidian) [reply .tool-versions]: Perhaps you can keep the file for you locally, but add it to [git global ignore config](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer)?

Comment (sergei-ivanov) [reply .tool-versions]: Yes, that's what I am planning to do, only on a local level (`.git/info/exclude`).

Comment (invidian) [reply .tool-versions]: Thanks, I appreciate the understanding :)

File: go.mod
Comment (invidian) [reply go.mod:3]: Hmm, yeah. I thought it won't compile with 1.16 when this is set to 1.17, but it seems to work, so it should be fine. Marking as resolved.

**Comment by invidian** [comment]
I've opened #12 to fix some basic stuff, let's wait for it to be merged before we merge this one.

**Comment by invidian** [comment]
#12 is now merged.

File: Dockerfile
Comment (sergei-ivanov) [inline Dockerfile:10]: I had a warning during docker build:
```
golangci/golangci-lint err this script is deprecated, please do not use it anymore. check https://github.com/goreleaser/godownloader/issues/207
```
The replacement was taken from the [official manual](https://golangci-lint.run/usage/install/#linux-and-windows).

File: go.mod
Comment (sergei-ivanov) [inline go.mod:7]: The latest version of this library is marked as deprecated, which breaks the linter. So I decided to keep the same version as before, and leave it for #9 to sort out crypto library upgrades.

**Review by sergei-ivanov — COMMENTED** [review]
OK, this should be ready now. Hopefully the build passes this time.

File: go.mod
Comment (invidian) [reply go.mod:7]: Ah, either this or I was thinking about suppressing the linter by a configuration line. Example: https://github.com/flexkube/libflexkube/blob/7717fd43362d6ca923f9d974485cf4af2b59893d/.golangci.yml#L14-L18

**Review by invidian — APPROVED** [review]
LGTM. Shall I merge @sergei-ivanov?

**Comment by sergei-ivanov** [comment]
> LGTM. Shall I merge @sergei-ivanov?

Yes please. Let's cut a new release, in order to unblock my colleagues on new shiny Macs. And after that we can start working on further improvements.

**Comment by invidian** [comment]
Thanks for help again. It's late today, I'll add some changelog and wrap a release tomorrow.

**Pull request merged**
